### PR TITLE
docs: add content

### PR DESCRIPTION
**Pull request by rlespinasse**

### CODE REVIEW

File: docs/markdown/40-reuse/00-reusable-workflow.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        action_input2: ${{ inputs.action_secret }}
+      env:
+        ACTION_ENV: ${{ env.WORKFLOW_ENV }}
+```
+
+##--##
+
+# TODO Reusable Workflows
+
+- Only actions can be call from both
```
Comment (rlespinasse) [inline docs/markdown/40-reuse/00-reusable-workflow.md:86]: Some changes about that
- https://github.blog/changelog/2022-08-22-github-actions-improvements-to-reusable-workflows-2/

Comment (rlespinasse) [reply docs/markdown/40-reuse/00-reusable-workflow.md:86]: > You can connect up to four levels of workflows.

File: docs/markdown/40-reuse/00-create.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<!-- .slide: -->
+
+# Create an action
+
+You can create an action from scratch based on an idea or you can extract some steps from a workflow and turn it into an action.
+Not everything should be an action but anything could became an action.
+
+When creating an action is in order, the question you can ask you is
```
Comment (ThibDujardin) [inline docs/markdown/40-reuse/00-create.md:8]: `When creating an action is in order`not sure about this sentence

File: docs/markdown/10-code/00-definition.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  - Security scan
+- ...
+
+##--##
+
+# Type of actions
+
+- JavaScript actions ([**template JS** 🔗](https://github.com/actions/javascript-action), [**template TS** 🔗](https://github.com/actions/typescript-action))
+- Docker container actions ([**template** 🔗](https://github.com/actions/container-action))
+- Composite actions
```
[the remark this answers is not in the document]
Comment (rlespinasse) [reply docs/markdown/10-code/00-definition.md:35]: Will add it on limitations slide

File: docs/markdown/10-code/01-TITLE.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<!-- .slide: class="transition bg-white" -->
+
+# Action anatomy
```
[the remark this answers is not in the document]
Comment (rlespinasse) [reply docs/markdown/10-code/01-TITLE.md:3]: We will keep the term structure as proposed

File: docs/markdown/10-code/01-inputs.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+name: Action name
+description: Short description of the action
+runs:
+  using: 'node16'
+  main: 'dist/main.js'
+inputs:
+  firstInput:
+    description: First Input
+    required: true
+    default: "Default value"
```
[the remark this answers is not in the document]
Comment (rlespinasse) [reply docs/markdown/10-code/01-inputs.md:19]: fine by me 👍

File: docs/markdown/10-code/01-outputs.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  someOutput:
+    description: "Some Output"
+```
+
+To define an output value, you need to print the `set-output` command
+
+```text
+::set-output name=someOutput::Some Value
+```
+
```
[the remark this answers is not in the document]
Comment (rlespinasse) [reply docs/markdown/10-code/01-outputs.md:53]: I think it's not necessary at this section. But we keep use the idea of telling the time on the exercice by exemple.

File: docs/markdown/40-reuse/00-create.md (hunk 2 above)
Comment (rlespinasse) [reply docs/markdown/40-reuse/00-create.md:8]: Rephrase, tell me what you think of it.

File: docs/markdown/10-code/00-definition.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rlespinasse) [reply docs/markdown/10-code/00-definition.md:38]: Already done in the virtual environment limitation slide

**Comment by ThibDujardin** [comment]
We don't speak about the secret / specific secret per env 
Could it be interesting in this school?

**Comment by rlespinasse** [comment]
> We don't speak about the secret / specific secret per env Could it be interesting in this school?

What are the specifics about them? My first thought is more to tell about this in the other school, but this can have an impact to take in account on the `reuse` section with reusable workflow vs action.

**Pull request closed without merging**